chain-crypto/src/algorithms/sumed25519/mod.rs

```diff
@@ -115,3 +115,30 @@ impl KeyEvolvingSignatureAlgorithm for SumEd25519_12 {
         sig.t() as u32
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    /// `public_from_binary` and `signature_from_bytes` should fail if the provided byte array does
+    /// not match the public key size
+    fn sum_ed25519_12_size_checks() {
+        for n in 0..u8::MAX as usize {
```

Comment:
as discussed, let's try to rewrite test so it uses arbitrary for size of input vector

Reply:
Done.